### PR TITLE
Add description of `ValueError` and `TypeErorr` to `Raises` section of `Trial.report` 

### PR DESCRIPTION
**Pull request by MasahitoKumada**
## Motivation

Related to #1618.

## Description of the changes

Add `Raises of ValueError & TypeError` in `optuna.trial._trial.report`.

### CODE REVIEW

File: optuna/trial/_trial.py [hunk 1]
```diff
@@ -127,7 +127,12 @@ def objective(trial):
             low:
                 Lower endpoint of the range of suggested values. ``low`` is included in the range.
             high:
-                Upper endpoint of the range of suggested values. ``high`` is included in the range.
+                Upper endpoint of the range of suggested values. ``high`` is excluded from the
+                range.
+
+                .. note::
+                    If ``step`` is specified, ``high`` is included as well as ``low``.
+
```
Comment (toshihikoyanase) [inline optuna/trial/_trial.py:135]: I don't think this change is related to the PR title. Could you revert it, please?

**Review by toshihikoyanase — CHANGES_REQUESTED** [review]
This PR has several changes that are not related to `Trial.report`. Could you revert such changes?

File: optuna/trial/_trial.py [hunk 2]
```diff
@@ -469,6 +469,10 @@ def objective(trial):
         Raises:
             :exc:`NotImplementedError`:
                 If trial is being used for multi-objective optimization.
+            :exe:`ValueError`:
+                If ``step is not None`` is specified.
```
Comment (toshihikoyanase) [inline optuna/trial/_trial.py:473]: In my understanding, `ValueError` is raised when a negative number is specified to `step`. Could you explain where did you find `None` value regarding ` ValueError`?

File: optuna/trial/_trial.py [hunk 3]
```diff
@@ -469,6 +469,10 @@ def objective(trial):
...
+            :exe:`TypeError`:
+                If ``value of type but a float`` is specified.
```
Comment (toshihikoyanase) [inline optuna/trial/_trial.py:475]: I guess this sentence is a bit fragmented. In my understanding, the `TypeError` is raised when the type of `value` is not float-castable. Could you rephase it, for example, as follows?

```suggestion
            :exe:`TypeError`:
                If the type of ``value`` is not obj:`float`.
```

**Review by toshihikoyanase — COMMENTED** [review]
Thank you for your update. I have two small comments.

File: optuna/trial/_trial.py [hunk 4]
```diff
@@ -469,6 +469,10 @@ def objective(trial):
         Raises:
             :exc:`NotImplementedError`:
                 If trial is being used for multi-objective optimization.
+            :exe:`ValueError`:
+                If ``step is negative`` is specified.
```
Comment (toshihikoyanase) [inline optuna/trial/_trial.py:473]: How about simplifying the sentence?
```suggestion
                If ``step`` is negative.
```

File: optuna/trial/_trial.py (hunk 3 above)
Comment (MasahitoKumada) [reply optuna/trial/_trial.py]: I also think I should revise the wording that way.

File: optuna/trial/_trial.py (hunk 4 above)
Comment (MasahitoKumada) [reply optuna/trial/_trial.py]: I also think I should revise the wording that way.

File: optuna/trial/_trial.py [hunk 5]
```diff
@@ -469,6 +469,10 @@ def objective(trial):
         Raises:
             :exc:`NotImplementedError`:
                 If trial is being used for multi-objective optimization.
+            :exe:`ValueError`:
+                If ``step`` is negative.
+            :exe:`TypeError`:
+                If the type of ``value`` is not obj:`float`.
```
Comment (toshihikoyanase) [inline optuna/trial/_trial.py:475]: Ah, I'm sorry but I missed `:`. The current rendering result is as follows:

image

```suggestion
                If the type of ``value`` is not :obj:`float`.
```

Comment (MasahitoKumada) [reply optuna/trial/_trial.py:475]: Thank you, permission to modify.

**Review by toshihikoyanase — APPROVED** [review]
Thank you for your update. Looks good to me.

**Comment by toshihikoyanase** [comment]
Let me update the title to improve the readability of the release note.

**Review by nzw0301 — APPROVED** [review]
LGTM!

**Pull request merged**